cmd/kubeless/update.go

```diff
@@ -121,14 +122,35 @@ var updateCmd = &cobra.Command{
 			}
 			funcDeps = string(bytes)
 		}
+		var headless *bool = nil
+		var port *int32 = nil
+		cmd.Flags().Visit(func(flag *pflag.Flag) {
+			switch flag.Name {
+			case "headless":
+				val, err := cmd.Flags().GetBool("headless")
+				headless = &val
+				if err != nil {
+					logrus.Fatal(err)
+				}
+			case "port":
+				val, err := cmd.Flags().GetInt32("port")
+				port = &val
+				if err != nil {
+					logrus.Fatal(err)
+				}
+			}
+		})
 
 		previousFunction, err := utils.GetFunction(funcName, ns)
 		if err != nil {
 			logrus.Fatal(err)
 		}
 
+		if port != nil && (*port <= 0 || *port > 65535) {
```

Comment:
I would set this check in the `getFunctionDescription` function so you don't need to repeat it twice